addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/ParamSinksUtils.java

```diff
@@ -188,17 +156,45 @@ private static String getCachedItem(Map<String, String> map, String item) {
 
     private static class UserDataSource {
 
+        private static final String GENERIC_STRING = "DYX";
         private final String uri;
         private final String param;
         private final String stringRepresentation;
 
         public UserDataSource(HttpMessage sourceMsg, String param) {
             super();
-            this.uri = getCachedItem(cachedUris, sourceMsg.getRequestHeader().getURI().toString());
+
+            this.uri =
+                    getCachedItem(
+                            cachedUris,
+                            createGenericUri(sourceMsg.getRequestHeader().getURI(), param));
             this.param = getCachedItem(cachedParams, param);
             this.stringRepresentation = uri + "#" + param;
         }
 
+        String createGenericUri(URI uri, String param) {
+            String uriString = uri.toString();
+            // if the parameter is in the path we need to replace it with something more abstract
+            // otherwise
+            // we will not be able to find sinks when we change the value of the parameter
+            String path = uri.getEscapedPath();
```

Comment:
@DiogoMRSilva  can you please explain this part where we are replacing the param with generic string as in which case this can be a problem that we are replacing it with generic string and also if there can be an issue then why we are not doing the same at line 172 as the same thing can be an issue while retrieving. I am struggling to understand this.

Also my feeling is the key will not change only the value will change and we are passing the key to this method.

thanks,
Karan

Reply:
For param value i understand the reason, not able to understand the path param usecase

Reply:
Thx for the review Karan.
> why we are not doing the same at line 172

You are correct, the string representation will not be generic. I need to change that logic to make param also generic.

> which case this can be a problem

It will be a problem in any case you are testing a path param and you retrieve sinks with a changed request. For example you register a sink for a path param `setSinkForSource( sourceMsg = POST /create/diogo, param = diogo, sinkMsg = GET /admin/users)` , if you are testing XSS and you ask for the sinks like `getSinksIdsForSource(sourceMsg = POST /create/<script>, param = diogo)`  you will not get the sinks you should receive because the `stringRepresentation` will not match.